.github/workflows/build-vscode-extension.yml

```diff
@@ -35,35 +46,106 @@ jobs:
-        working-directory: vscode/microsoft-kiota
-      - name: run tests
-        if: runner.os != 'Linux'
-        run: npm run test:coverage
-        working-directory: vscode/microsoft-kiota
+        run: |
+          npm install
+          npm list
+        working-directory: vscode
+
+      - name: Lint
+        run: npm run lint
+        working-directory: vscode
+
+      - name: run tests(linux) - ${{ matrix.builds.id }}
+        if: runner.os == 'Linux' 
+        run: xvfb-run -a npm run test:${{ matrix.builds.id }}:coverage
+        working-directory: vscode
+
+      - name: run tests - ${{ matrix.builds.id }}
+        if: runner.os != 'Linux' 
+        run: npm run test:${{ matrix.builds.id }}:coverage
+        working-directory: vscode
+
       - name: Run sonar cloud analysis
         if: needs.checksecret.outputs.is_SONAR_TOKEN_set == 'true'
         uses: SonarSource/sonarqube-scan-action@v5
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         with:
-          projectBaseDir: vscode/microsoft-kiota
-      - run: npm run package
-        working-directory: vscode/microsoft-kiota
-      - run: npm i -g @vscode/vsce
-      - run: vsce package
-        working-directory: vscode/microsoft-kiota
+          projectBaseDir: vscode
+          args: >
+            -Dsonar.javascript.lcov.reportPaths=coverage/lcov.info,../npm-package/coverage/lcov.info
+
+      - run: npm run package:${{ matrix.builds.id }}
+        if: matrix.builds.id == 'package'
+        name: 'Package - ${{ matrix.builds.id }}'
+        working-directory: vscode
+
+      - run: |
+          npm i -g @vscode/vsce
+          vsce package
+        working-directory: vscode/${{ matrix.builds.path }}
+        name: 'Package - ${{ matrix.builds.id }}'
+        if: matrix.builds.id == 'vscode'
+
+        # back to common operations
+
       - name: Upload artifact
         uses: actions/upload-artifact@v4
         with:
-          name: vscode-extension
-          path: vscode/microsoft-kiota/*.vsix
+          name: ${{ matrix.builds.name }}
+          path: vscode/${{ matrix.builds.path }}/*.${{ matrix.builds.item }}
+
+  generate_binaries:
+      needs: [checksecret]
+      strategy:
+        matrix:
+          binaries:
+            - architecture: win-x86
+              os: windows-latest
+            - architecture: win-x64
+              os: windows-latest
+            - architecture: linux-x64
+              os: ubuntu-latest
+            - architecture: linux-arm64
+              os: ubuntu-latest
+            - architecture: osx-x64
+              os: macOS-latest
+            - architecture: osx-arm64
+              os: macOS-latest
+      runs-on: ${{ matrix.binaries.os }}
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v4
+  
+        - name: Set up Node.js
+          uses: actions/setup-node@v4
+          with:
+            node-version: 20.x
+  
+        - name: Use .NET 6
```

Comment:
net6 is only required for ESRP sign, which we're not using here. I think we can remove that step